vortex-array/src/compute/scalar_subtract.rs

```diff
@@ -0,0 +1,131 @@
+use vortex_error::{vortex_err, VortexResult};
+
+use crate::scalar::Scalar;
+use crate::{Array, OwnedArray};
+
+pub trait ScalarSubtractFn {
+    fn scalar_subtract(&self, to_subtract: Scalar) -> VortexResult<OwnedArray>;
+}
+
+pub fn scalar_subtract(array: &Array, to_subtract: Scalar) -> VortexResult<OwnedArray> {
+    array.with_dyn(|c| {
+        let option = c
+            .scalar_subtract()
+            .map(|t| t.scalar_subtract(to_subtract.clone()));
```

Comment:
I think you can avoid this clone if you mark your closures with e.g. `move |c|`. Not totally sure

Reply:
Move doesn't quite work here, but it looks like passing a ref instead makes everything nice.